src/controllers/product.ts

```diff
@@ -0,0 +1,47 @@
+import {Request, Response} from "express";
+import {getUserById} from "../services/user";
+import {returnUnauthorized} from "../utils/return-unauthorized";
+import {ID_HEADER_NAME} from "./user";
+import {getProduct, getProducts} from "../services/product";
+
+export const productsHandler = (req: Request, res: Response) => {
+  const id = req.get(ID_HEADER_NAME)
+  const user = id && getUserById(id);
+
+
+  if (!user) {
+    returnUnauthorized(res)
+    return;
+  }
+
+  const result = getProducts()
+  res
+      .status(200)
+      .send({data: result, error: null});
+}
+
+export const productHandler = (req: Request, res: Response) => {
+  const id = req.get(ID_HEADER_NAME)
+  const user = id && getUserById(id);
+
+
+  if (!user) {
+    returnUnauthorized(res)
+    return;
+  }
```

Comment:
You check authorization in your middleware (it will be expanded in further lessons I believe), so all errors regarding auth will be resolved in said middleware.